### PR TITLE
finish wiring PrintFlags

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Finish wiring PrintFlags through commands

cc @deads2k @soltysh

### CODE REVIEW

File: pkg/kubectl/cmd/rollingupdate.go [hunk 1]
```diff
@@ -138,8 +164,38 @@ func validateArguments(cmd *cobra.Command, filenames, args []string) error {
 	return utilerrors.NewAggregate(errors)
 }
 
-func RunRollingUpdate(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *resource.FilenameOptions) error {
-	err := validateArguments(cmd, options.Filenames, args)
+func (o *RollingUpdateOptions) Complete(cmd *cobra.Command, args []string) error {
+	o.OldName = args[0]
+
+	o.DeploymentKey = cmdutil.GetFlagString(cmd, "deployment-label-key")
```
Comment (dixudx) [inline pkg/kubectl/cmd/rollingupdate.go]: You can bind these flags to struct `RollingUpdateOptions`, which will be easily unit tested with predictable default behavior. #60363 FYI.

/cc @soltysh

**Review by dixudx — CHANGES_REQUESTED** [review]

File: pkg/kubectl/cmd/rollingupdate.go [hunk 2]
```diff
@@ -85,23 +109,25 @@ func NewCmdRollingUpdate(f cmdutil.Factory, out io.Writer) *cobra.Command {
 		Long:    rollingUpdateLong,
 		Example: rollingUpdateExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunRollingUpdate(f, out, cmd, args, options)
-			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Complete(cmd, args))
+			cmdutil.CheckErr(o.Run(f, out, cmd, args))
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollingupdate.go]: Get rid of factory from `Run` at all costs!

File: pkg/kubectl/cmd/rollout/rollout_pause.go [hunk 3]
```diff
@@ -63,7 +66,10 @@ var (
 )
 
 func NewCmdRolloutPause(f cmdutil.Factory, out io.Writer) *cobra.Command {
-	options := &PauseConfig{}
+	o := &PauseConfig{
+		PrintFlags: printers.NewPrintFlags("paused"),
+		Out:        out,
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollout/rollout_pause.go]: Use `IOStreams`.

File: pkg/kubectl/cmd/rollout/rollout_resume.go [hunk 4]
```diff
@@ -61,7 +64,10 @@ var (
 )
 
 func NewCmdRolloutResume(f cmdutil.Factory, out io.Writer) *cobra.Command {
-	options := &ResumeConfig{}
+	o := &ResumeConfig{
+		PrintFlags: printers.NewPrintFlags("resumed"),
+		Out:        out,
```
Comment (soltysh) [inline pkg/kubectl/cmd/rollout/rollout_resume.go]: Use `IOStreams`.

File: pkg/kubectl/cmd/scale.go [hunk 5]
```diff
@@ -64,21 +65,27 @@ var (
 type ScaleOptions struct {
 	FilenameOptions resource.FilenameOptions
 	RecordFlags     *genericclioptions.RecordFlags
+	PrintFlags      *printers.PrintFlags
+	PrintObj        printers.ResourcePrinterFunc
 
 	Recorder genericclioptions.Recorder
+
+	Out io.Writer
```
Comment (soltysh) [inline pkg/kubectl/cmd/scale.go]: `IOStreams`

File: pkg/kubectl/cmd/taint.go [hunk 6]
```diff
@@ -79,7 +83,10 @@ var (
 )
 
 func NewCmdTaint(f cmdutil.Factory, out io.Writer) *cobra.Command {
-	options := &TaintOptions{}
+	options := &TaintOptions{
+		PrintFlags: printers.NewPrintFlags("tainted"),
+		out:        out,
```
Comment (soltysh) [inline pkg/kubectl/cmd/taint.go]: `IOStreams`

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by juanvallejo** [comment]
@soltysh thanks for the review, comments addressed

**Review by soltysh — APPROVED** [review]
/approve no-issue

You just need to update bazel and this is good to go.

**Comment by juanvallejo** [comment]
@soltysh thanks, tests are now passing

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by deads2k** [comment]
/hold

conflicts with restmapping again.

**Comment by soltysh** [comment]
/hold cancel
the restmapping pr (#63105) merged

**Comment by soltysh** [comment]
Closed in favor of https://github.com/kubernetes/kubernetes/pull/63244

**Pull request closed without merging**